### DISCUSSION TITLE
Auto-create first row in Grid fields

### DISCUSSION THREAD

**Discussion by MatthewJLewis** [ExpressionEngine/ExpressionEngine#2987] [Feature Requests]
Requiring users to click "Add a Row" with Grid field types, when the grid doesn't yet contain any rows, seems counter-intuitive to many of the authors I work with. Wouldn't it make sense for the UI to automatically assume that a new row is to be added and display the fields by default?

At the very least, I'd like to see this behavior when Grid fields are included in Fluid fields, where it could be safely assumed the author intends to create at least one row or they wouldn't have chosen to Add the field to begin with. :)

Alternatively, an auto-expand-first-row option could be available when creating or editing a Grid field.

**Comment by litzinger** [comment]
I would disagree with this for standalone Grid fields. Auto-generating a first blank row in the UI for the user implies that the field is required. If a blank row were added to the UI, but no data submitted, the backend would need to then remove that row, otherwise it may insert a blank row in the DB, which isn't good for a normalized dataset.

I can see the argument for this behavior when adding a Grid as a Fluid field... the user has already taken the action of adding a Grid to the page, the second click to add the first Grid row is a redundant action.

**Comment by MatthewJLewis** [comment]
Good points for the stand alone grid fields, provided they are not, in
fact, required fields. If they are, then I still think auto-adding at least
the first row (if not the full minimum number of required rows) makes the
most sense.

However, from my experience working with authors who are not very tech
savvy, I think adding a display option or two to the grid field type would
provide the greatest flexibility. For instance, an option to "Auto add X
rows on new entry form" would make my life much easier.

**Comment by PDXMan1** [comment]
I think this would work well as a grid field option. "Create first grid row? Yes/No".  

That said, if it wasn't a required field, and it was empty on submission, it would be nice if an empty row wouldn't be saved as part of the entry. We run into unwanted empty rows now and then. We like that the first of several rows can be saved when empty, but if it's the only one, and left empty, then perhaps it doesn't get saved? Would that work for others?

**Comment by ic360-com** [comment]
I think the Auto Grid Row creation option would be a good idea.... especially where the Max Number of Grid Rows is set to 1.
Personally, when I'm editing/creating content... (with Grids as part of a Fluid field)... I'm bored of clicking "Add Row" whenever I add a new Grid field row to my Fluid field.

**Comment by shbchk** [comment]
Actually, this feature is already there. Just set "Minimum rows" to 1:

![image](https://github.com/ExpressionEngine/ExpressionEngine/assets/12234507/c5b890b0-7432-4beb-bc38-4a3268120120)

**Reply by ic360-com** [reply]
Good point... this is an option I hadn't thought of.
